fix(integrations): Add `fixes <SHORT-ID>` pitfalls with squashed commits (#5684)
getsentry/sentry-docs: src/docs/product/integrations/source-code-mgmt/github/index.mdx

You can also resolve issues with pull requests by including `fixes <SENTRY-SHORT-ID>` in the title or description.

> When Sentry sees this, we’ll automatically annotate the matching issue with a reference to the commit or pull request, and, later, when that commit or pull request is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release, if the commit is squashed Sentry will not know when the commit has been released and the issue may never be marked as a regression. With GitHub, we recommend using `fixes <SENTRY-SHORT-ID>` in the body of your pull request that is going to be merged to your main branch.

Reviewer: ```suggestion
When Sentry sees this, we’ll automatically annotate the matching issue with a reference to the commit or pull request, and, later, when that commit or pull request is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release. Otherwise, if the commit is squashed, Sentry won’t know when the commit has been released, and the issue may never be marked as a regression. With GitHub, we recommend using `fixes <SENTRY-SHORT-ID>` in the body of your pull request that is going to be merged to your main branch.
```
